File: ecomrs/the_real_talk/bad.c
int main() {
  // Initialize
  // Big array
  int arr[1000000];

  int i = 0;
  for (i = 0; i < 1000000; i++) {
    arr[i] = i;
  }

  // Map
  int map_arr[1000000];
  for (i = 0; i < 1000000; i++) {
    map_arr[i] = arr[i] * 2;
  }

  // Reduce
  int acc = 0;
  for (i = 0; i < 1000000; i++) {
    acc += arr[i];
  }

  // Select
  int filtered_arr[500000];
  for (i = 0; i < 1000000; i++) {
    if (i % 2 == 0) {
      filtered_arr[i / 2] = arr[i];
    }
  }

  return 0;
}

File: ecomrs/the_real_talk/bad.rb
class TestClass
  def initialize
    # Big array
    @array = []

    for i in (0..1000000)
      @array << i
    end
  end

  # Map
  def map
    map_arr = []
    @array.each do |x|
      map_arr[x] = @array[x] * 2
    end
    map_arr
  end

  # Reduce
  def reduce
    acc = 0;
    @array.each do |x|
      acc += x
    end
    acc
  end

  # Select
  def select
    filtered_arr = [];
    @array.each do |x|
      if x % 2 == 0
        filtered_arr << x
      end
    end
    filtered_arr
  end
end
